ndcube/extra_coords/extra_coords.py

```diff
@@ -0,0 +1,324 @@
+                raise ValueError(
+                    "Values in the mapping can not be larger than the number of pixel dimensions in the WCS."
+                )
+
+        self._mapping = mapping
+
+    @property
+    def wcs(self):
+        # docstring in ABC
+        if self._wcs is not None:
+            return self._wcs
+
+        if not self._lookup_tables:
+            return None
+
+        lutcs = set(lt[1] for lt in self._lookup_tables)
+        # created a sorted list of unique items
+        _tmp = set()  # a temporary set
+        lutcs = [x[1] for x in self._lookup_tables if x[1] not in _tmp and _tmp.add(x[1]) is None]
+        out = copy.deepcopy(lutcs[0])
+        for lut in lutcs[1:]:
+            out = out & lut
+        return out.wcs
+
+    @wcs.setter
+    def wcs(self, wcs):
+        if self._wcs is not None:
+            raise AttributeError(
+                "Can't set wcs if a WCS has already been specified."
+            )
+
+        if self._lookup_tables:
+            raise AttributeError(
+                "Can't set mapping manually when ExtraCoords is built from lookup tables."
+            )
+
+        if self._mapping is not None:
+            if not max(self._mapping) <= wcs.pixel_n_dim - 1:
+                raise ValueError(
+                    "Values in the mapping can not be larger than the number of pixel dimensions in the WCS."
+                )
+
+        self._wcs = wcs
+
+    def _getitem_string(self, item):
+        """
+        Slice the Extracoords based on axis names.
+        """
+
+        for names, lut in self._name_lut_map.items():
+            if item in names:
+                new_ec = ExtraCoords()
+                new_ec._lookup_tables = [lut]
+                return new_ec
+
+        raise KeyError(f"Can't find the world axis named {item} in this ExtraCoords object.")
+
+    def _getitem_lookup_tables(self, item):
+        """
+        Apply an array slice to the lookup tables.
+
+        Returns a new ExtraCoords object with modified lookup tables.
+        """
+        new_lookup_tables = set()
+        for lut_axis, lut in self._lookup_tables:
+            lut_axes = (lut_axis,) if not isinstance(lut_axis, tuple) else lut_axis
+            lut_slice = tuple(item[i] for i in lut_axes) if isinstance(item, tuple) else item
+
+            sliced_lut = lut[lut_slice]
+            if sliced_lut:
+                new_lookup_tables.add((lut_axis, sliced_lut))
+
+        new_extra_coords = type(self)()
+        new_extra_coords._lookup_tables = tuple(new_lookup_tables)
+        return new_extra_coords
+
+    def _getitem_wcs(self, item):
+        item = sanitize_slices(item, self.wcs.pixel_n_dim)
+
+        # It's valid to slice down the EC such that there is nothing left,
+        # which is not a valid way to slice the WCS
+        if len(item) == self.wcs.pixel_n_dim and all(isinstance(i, Integral) for i in item):
+            return type(self)()
```

Comment:
Doesn't slicing all axes with an int leave the coordinate values at the single pixel?  So there should be something left?  Or would these values get transferred to `GlobalCoords`?

Reply:
It would be copied into `GlobalCoords` when that's implemented.

Reply:
Am working on this now